packages/fluentui/react/src/components/Icon/SvgIcon.tsx

```diff
@@ -0,0 +1,146 @@
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+// @ts-ignore
+import { ThemeContext } from 'react-fela';
+
+import { UIComponentProps, ChildrenComponentProps, commonPropTypes, SizeValue } from '../../utils';
+import { /* FluentComponentStaticProps, */ ProviderContextPrepared, WithAsProp, withSafeTypeForAs } from '../../types';
+
+export type SvgIconXSpacing = 'none' | 'before' | 'after' | 'both';
+
+type SvgIconFuncArg = {
+  classes: { [iconSlot: string]: string };
+  rtl: boolean;
+  props: any; // TODO SvgIconProps
+};
+
+type SvgIconChildrenFn = (props: SvgIconFuncArg) => React.ReactNode;
+
+export interface SvgIconProps extends UIComponentProps, ChildrenComponentProps<SvgIconChildrenFn> {
+  /** Alternative text. */
+  alt?: string;
+  'aria-label'?: AccessibilityAttributes['aria-label'];
+
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility<IconBehaviorProps>;
+
+  /** SvgIcon can appear with rectangular border. */
+  bordered?: boolean;
+
+  /** SvgIcon can appear as circular. */
+  circular?: boolean;
+
+  /** An icon can show it is currently unable to be interacted with. */
+  disabled?: boolean;
+
+  /** An icon can provide an outline variant. */
+  outline?: boolean;
+
+  /** An icon can be rotated by the degree specified as number. */
+  rotate?: number;
+
+  /** Size of the icon. */
+  size?: SizeValue;
+
+  /** Adds space to the before, after or on both sides of the icon, or removes the default space around the icon ('none' value) */
+  xSpacing?: SvgIconXSpacing;
+}
+
+const SvgIcon: React.FC<WithAsProp<SvgIconProps>> & {
+  className: string;
+  handledProps: (keyof SvgIconProps)[];
+  create: (svg: SvgIconChildrenFn) => React.FC;
+} = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+
+  const { setStart, setEnd } = useTelemetry(SvgIcon.displayName, context.telemetry);
+  setStart();
+
+  const {
+    accessibility,
+    alt,
+    'aria-label': ariaLabel,
+    bordered,
+    circular,
+    className,
+    disabled,
+    design,
+    children,
+    outline,
+    rotate,
+    size,
+    styles,
+    variables,
+    xSpacing
+  } = props;
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: SvgIcon.displayName,
+    mapPropsToBehavior: () => ({
+      alt,
+      'aria-label': ariaLabel
+    }),
+    rtl: context.rtl
+  });
+  const { classes } = useStyles(SvgIcon.displayName, {
+    className: SvgIcon.className,
+    mapPropsToStyles: () => ({
+      bordered,
+      circular,
+      disabled,
+      outline,
+      rotate,
+      size,
+      xSpacing
+    }),
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl
+  });
+
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(SvgIcon.handledProps, props);
+
+  const element = (
+    <ElementType {...getA11Props('root', { className: classes.root, ...unhandledProps })}>
```

Comment:
Some icons may have more props, so unhandledProps may spread those on the root :\

Reply:
Started to think about [@emotion/is-prop-valid](https://www.npmjs.com/package/@emotion/is-prop-valid), however there we can add `handledProps` to the `.create()` factory